cmd/juju/application/bundle/bundle.go

```diff
@@ -345,3 +349,52 @@ func verifyBundle(data *charm.BundleData, bundleDir string, verifyConstraints fu
 	}
 	return errors.Trace(verifyError)
 }
+
+func verifyMixedSeriesBasesMatch(data *charm.BundleData) error {
+	if data.Series != "" && data.DefaultBase != "" {
+		b, err := corebase.ParseBaseFromString(data.DefaultBase)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		s, err := corebase.GetSeriesFromBase(b)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if s != data.Series {
+			return errors.NotValidf("supplied series %q and base %q in bundle default do not match", data.Series, data.DefaultBase)
```

Comment:
I'd say something like, `errors.NewNotValid(nil, "bundle series %q and base %q must match if supplied together")`